File: leetcode/001 Two Sum/main.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numshash = {}
        i = 0

        while i < len(nums):
            sum = target - nums[i]
            if sum in numshash:
                return [i, numshash[sum]]
            numshash[nums[i]] = i
            i += 1
            

File: leetcode/001 Two Sum/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function(nums, target) {
    let numshash = {};

    for(let i = 0; i < nums.length; i++) {
        let sum = target - nums[i];

        if(numshash.hasOwnProperty(String(sum))) {
            return[i, numshash[String(sum)]];
        }
        
        numshash[String(nums[i])] = i;
    }
};